### PR TITLE
[WIP] Added memcheck to the build matrix

### PR DESCRIPTION
**Pull request by sergeyklay**
Related: #382

### CODE REVIEW

**Comment by sergeyklay** [comment]
@ysmolsky I would like to know should we suppress some errors or not

**Comment by ysmolsky** [comment]
> @ysmolsky I would like to know should we suppress some errors or not

I would prefer not to see "still reachable" kind of errors. The most important are "definitely lost" and "possibly lost". We need to pay attention to these first.

http://valgrind.org/docs/manual/faq.html#faq.deflost

File: .travis.yml [hunk 1]
```diff
@@ -77,6 +81,27 @@ jobs:
...
+    - stage: Memcheck
+      os: linux
+      compiler: gcc
+      env: MEMCHECK=1
+      before_script:
+          - export COVERAGE=OFF
+      script:
+          - make $MAKEJOBS
+          - make $MAKEJOBS ct/_ctcheck
+          - export MAKEFLAGS="valgrind --leak-check=full"
```
Comment (ysmolsky) [inline .travis.yml]: We can get rid of it by updating to the last version of "ct". I have fixed it there. Let me create patch.

File: .travis.yml [hunk 2]
```diff
@@ -77,6 +81,27 @@ jobs:
...
+          - |
+            valgrind \
+                --read-var-info=yes \
+                --error-exitcode=1 \
+                --fullpath-after= \
+                --track-origins=yes \
+                --leak-check=full \
+                --show-leak-kinds=all \
```
Comment (ysmolsky) [inline .travis.yml]: Please see my comment about what kind of stuff we do not need to show.

File: .travis.yml (hunk 1 above)
Comment (ysmolsky) [reply .travis.yml]: PR was created. After the merge we can remove the (kludge) line above.

**Comment by ysmolsky** [comment]
I am not sure what to do with memory leaks. After all we decided not to fix them in the program, which is supposed to be crash-restarted. @sergeyklay any ideas?

**Comment by sergeyklay** [comment]
@ysmolsky I'm sorry for the delay (too busy at work). Can this wait for some (short) time? I'll try to investigate a bit more in the near future.

**Comment by ysmolsky** [comment]
> @ysmolsky I'm sorry for the delay (too busy at work). Can this wait for some (short) time? I'll try to investigate a bit more in the near future.

Sure thing. By all means, no rushing from me.

**Comment by ysmolsky** [comment]
I think, we won't go there yet. I have somewhat reduced interest in this project since autumn has started. I was a little depressed that there is very little support by the community.  What was done is good enough for my work and I will fix only critical bugs. Most likely I will not try to hunt down all kind of memory leaks since this software is intended as crash-only and there is no sense in graceful exit.

**Comment by sergeyklay** [comment]
Ok, let's close this one :)

**Pull request closed without merging**